### PR TITLE
Add spec test and basic fixture for YAML format

### PR DESCRIPTION
**Pull request by andressrg**
WIP for discussion

### CODE REVIEW

File: test/yamlContext/fixtures/basic.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  fb-info: 'Info to show to facebook crawler'
+  twitter-info: 'Info that twitter crawler grabs'
+
+
+# pages names match hbs files
+pages:
+  home:
+    needs:  # represent api fetches
+      # categories is special, they map to the categories base key in this yaml file
+      categories: [
```
Comment (tbarho) [inline test/yamlContext/fixtures/basic.yaml]: should we call this something more consistent with api, like `groups`?

Comment (tbarho) [inline test/yamlContext/fixtures/basic.yaml]: oh, i should have read the whole thing... =)

File: test/yamlContext/fixtures/basic.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        call: 'payments.list'
+        as: 'payments'
+
+      url:
+        path: 'http://api.github.com/pulls?user=take2'
+        as: 'pull-requests’
+
+  freeway-signs:
+    needs:      
+      # default behavior of categories
```
Comment (tbarho) [inline test/yamlContext/fixtures/basic.yaml:39]: i think this maybe was copied?

Comment (tbarho) [inline test/yamlContext/fixtures/basic.yaml:39]: oh no i understand now.  you're just illustrating the default.  that could be removed with the same effect

Comment (andressrg) [inline test/yamlContext/fixtures/basic.yaml:39]: yes, the adapter part

**Pull request merged**